frontend/src/scenes/pipeline/hogfunctions/hogFunctionConfigurationLogic.tsx

```diff
@@ -1135,7 +1137,37 @@ export const hogFunctionConfigurationLogic = kea<hogFunctionConfigurationLogicTy
     })),
 
     beforeUnload(({ values, cache }) => ({
-        enabled: () => !cache.disabledBeforeUnload && !values.unsavedConfiguration && values.configurationChanged,
+        enabled: (newLocation?: CombinedLocation) => {
+            if (cache.disabledBeforeUnload || values.unsavedConfiguration || !values.configurationChanged) {
+                return false
+            }
+
+            // the oldRoute includes the project id, so we remove it for comparison
+            const oldRoute = router.values.location.pathname.replace(/\/project\/\d+/, '').split('/')
+            const newRoute = newLocation?.pathname.replace(/\/project\/\d+/, '').split('/')
+
+            if (!newRoute || newRoute.length !== oldRoute.length) {
```

Comment:
When I said it was flakey - these are the bits im referring to. Doing a bunch of pathname replace and splitting. Feels like this is the kind of flakey code that will break in ways that are hard to validate in the future.

It's also based on a bunch of known knowledge - like you "know" that the testing tab uses this logic which is why you have it here but what if the Testing page later doesn't use it? Would you account for other pages that _do_ use this logic? 

I can approve to unblock but I'd love a much more reliable solution here, I just don't know what that is tbh...